sgpt/cache.py

```diff
@@ -84,16 +85,24 @@ def __call__(self, func: Callable) -> Callable:
         """
         def wrapper(*args, **kwargs):
             chat_id = kwargs.pop("chat_id", None)
-            message = {"role": "user", "content": kwargs.pop("message")}
+            # system messages stuff
+            system_kwarg = kwargs.pop("system", None)
+            if system_kwarg is not None:
+                system_message = {"role": "system", "content": system_kwarg}
+            else:
+                system_message = {"role": "system", "content": config.get("DEFAULT_SYSTEM_MESSAGE")}
+            message = {"role": "user", "content": kwargs.pop("messages")}
             if not chat_id:
-                kwargs["message"] = [message]
+                kwargs["messages"] = [system_message, message]
                 return func(*args, **kwargs)
-            kwargs["message"] = self._read(chat_id)
-            kwargs["message"].append(message)
-            response_text = func(*args, **kwargs)
-            kwargs["message"].append({"role": "assistant", "content": response_text})
-            self._write(kwargs["message"], chat_id)
-            return response_text
+            else:
+                kwargs["messages"] = self._read(chat_id)
+                kwargs["messages"].append(system_message)
```

Comment:
Check if chat session file not eixsts with given `chat_id` and add system_role just once for initianting a chat session. This is important since after couple of messages we will be passing redundant payloads with useles repeating system role messages. It adds latency to request + response time and OpenAI will spend time figuring out all these role messages. If user passed `--role` with `--chat {id}`:
```python
raise typer.BadParameter("Cannot use --role with initiated chat session.")
```

Reply:
I like the idea of `raise`ing in the event of a user trying to add another system role message. That is a good solution. Because --shell and --code are also moving to system role messages the error prompt can't be that simple though.